boards/FSM/Src/App/App_SetPeriodicCanSignals.c

```diff
@@ -143,7 +147,69 @@ void App_SetPeriodicSignals_AcceleratorPedal(const struct FsmWorld *world)
 
 void App_SetPeriodicSignals_MotorShutdownFaults(const struct FsmWorld *world)
 {
-    struct FsmCanTxInterface *can_tx = App_FsmWorld_GetCanTx(world);
+    struct FsmCanTxInterface *can_tx      = App_FsmWorld_GetCanTx(world);
+    struct FsmCanRxInterface *can_rx      = App_FsmWorld_GetCanRx(world);
+    static uint8_t            error_count = 0;
+
+    // Perform torque plausibility check if DCM is in the drive state
+    if (App_CanRx_DCM_STATE_MACHINE_GetSignal_STATE(can_rx) ==
+        CANMSGS_DCM_STATE_MACHINE_STATE_DRIVE_CHOICE)
+    {
+        const bool is_regen_allowed =
+            App_CanTx_GetPeriodicSignal_LEFT_WHEEL_SPEED(can_tx) >
+                REGEN_WHEEL_SPEED_THRESHOLD_KPH &&
+            App_CanTx_GetPeriodicSignal_RIGHT_WHEEL_SPEED(can_tx) >
+                REGEN_WHEEL_SPEED_THRESHOLD_KPH;
+        const float regen_paddle_percentage =
+            (float)App_CanRx_DIM_REGEN_PADDLE_GetSignal_MAPPED_PADDLE_POSITION(
+                can_rx);
+        float expected_torque_request;
+
+        if (regen_paddle_percentage > 0.0f && is_regen_allowed)
+        {
+            expected_torque_request =
+                -0.01f * regen_paddle_percentage * MAX_TORQUE_REQUEST_NM;
+        }
+        else
+        {
+            expected_torque_request =
```

Comment:
Just a general comment - I think we may need to account for the delay of the system here. It will take some time for the FSM to send the pedal CAN message to the DCM, and for the DCM to turn that value into a torque request, and then the FSM needs to receive the DCM torque request. FSM will calculate the maximum bound much faster than this transaction, so the value from the DCM that it's checking against will be out of date. Maybe we want to take the pedal value read ~10ms (or more) ago or something to mitigate this. I think the margins of error we're allowing could also account for this, but it's something to consider if it's throwing this fault when we release the accel pedal super fast.

Reply:
Sorry for the late reply. I was thinking this exact same thing. I placed an error counter as a way to mitigate this but it might be short-sighted

Reply:
Releasing the accel pedal will definitely be the worst-case scenario, will have to test for that.